aries_vcx/src/protocols/proof_presentation/verifier/states/finished.rs

```diff
@@ -1,15 +1,37 @@
-use crate::protocols::proof_presentation::verifier::state_machine::RevocationStatus;
+use serde::{Deserialize, Deserializer};
+
 use messages::concepts::problem_report::ProblemReport;
 use messages::protocols::proof_presentation::presentation::Presentation;
 use messages::protocols::proof_presentation::presentation_request::PresentationRequest;
 use messages::status::Status;
 
+use crate::protocols::proof_presentation::verifier::verification_status::PresentationVerificationStatus;
+
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 pub struct FinishedState {
     pub presentation_request: Option<PresentationRequest>,
     pub presentation: Option<Presentation>,
     pub status: Status,
-    pub revocation_status: Option<RevocationStatus>,
+    #[serde(
+        default = "verification_status_unavailable", // todo: to be removed in 0.54.0, this supports legacy serialization when the field was 'null'
+        deserialize_with = "null_to_unavailable" // todo: to be removed in 0.54.0, this supports legacy serialization when the field was 'undefined'
```

Comment:
This is unnecessary. `serde` will use the `default` impl if the value is not found. Or, more accurately, it will never resort to `default` because the field's absence is treated in the custom deserialize function.

Reply:
It seems to be necessary, the behaviour is different when value is `null` and `undefined`. 

The `default` is used if is the field is undefined completely in the json.
But if `verification_status` is present and is `null`, without `#[serde(deserialize_with = "null_to_unavailable")]` it will fail tests
```
 thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("expected value", line: 1, column: 89)', aries_vcx/src/protocols/proof_presentation/verifier/states/finished.rs:170:80
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
FAILED
test protocols::proof_presentation::verifier::states::finished::unit_tests::test_verifier_state_finished_deser_legacy_values_verification_status ... thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("expected value", line: 1, column: 91)', aries_vcx/src/protocols/proof_presentation/verifier/states/finished.rs:144:80
FAILED
```

To save you look ups, line 170:
```
let serialized =
                r#"{"presentation":null,"presentation_request":null,"status":"Success","revocation_status":null}"#;
            let deserialized: FinishedState = serde_json::from_str(serialized).unwrap();
```
line 144
```
            let serialized =
                r#"{"presentation":null,"presentation_request":null,"status":"Success","verification_status":null}"#;
            let deserialized: FinishedState = serde_json::from_str(serialized).unwrap();
```

Reply:
Fair point. But I believe that still makes the `default` redundant, as deserialization will just resort to `None` in the `deserialize` function and be handled accordingly.

Reply:
fair enough, removed `default`